tests/test_fieldset.py

```diff
@@ -242,3 +247,23 @@ class MyParticle(ptype[mode]):
     elif dt_sign == -1:
         temp_theo = temp_vec[0]
     assert np.allclose(temp_theo, pset.particles[0].temp, atol=1e-5)
+
+
+def test_fieldset_defer_loading_with_diff_time_origin(tmpdir, filename='test_parcels_defer_loading'):
```

Comment:
Don't we want to have an `xfail` for pytest when the two `time_origins` _don't_ match? This would really catch #351

Reply:
I did the mirror test.
In #351, two different `time_origin` were making the test pass while it was supposed to crash.
Here I defined two time vectors such that it crashes if the bug is still there.
I'll add the other config as well (it is a different time_vector with a different time_origin) to show both options.